homeassistant/components/surepetcare/device_tracker.py

```diff
@@ -0,0 +1,76 @@
+"""Support for Sure PetCare pet device tracking."""
+
+from __future__ import annotations
+
+from typing import cast
+
+from surepy.entities import SurepyEntity
+from surepy.entities.pet import Pet as SurepyPet
+from surepy.enums import EntityType, Location
+
+from homeassistant.components.device_tracker import TrackerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_HOME, STATE_NOT_HOME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import SurePetcareDataCoordinator
+from .entity import SurePetcareEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Sure PetCare device tracker entities based on a config entry."""
+    coordinator: SurePetcareDataCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities: list[SurePetcareDeviceTracker] = [
+        SurePetcareDeviceTracker(surepy_entity.id, coordinator)
+        for surepy_entity in coordinator.data.values()
+        if surepy_entity.type == EntityType.PET
+    ]
+
+    async_add_entities(entities)
+
+
+class SurePetcareDeviceTracker(SurePetcareEntity, TrackerEntity):
+    """Sure Petcare pet device tracker."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        surepetcare_id: int,
+        coordinator: SurePetcareDataCoordinator,
+    ) -> None:
+        """Initialize the pet device tracker."""
+        super().__init__(surepetcare_id, coordinator)
+        self._attr_unique_id = f"{self._device_id}-tracker"
```

Comment:
the unique id is unique per platform per intergation, so we don't have to append `tracker`